src/components/AuthRoute.tsx

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable @typescript-eslint/no-extra-non-null-assertion */
+import { useAuthState } from "react-firebase-hooks/auth"
+import { Navigate } from "react-router-dom"
+import { auth } from "../authConfig"
+import { FC } from "react";
+
+interface Props {
+    children: React.ReactNode;
+}
+
+export const AuthRoute: FC<Props> = ({ children }) => {
```

Comment:
```suggestion
export const AuthRoute: FC = ({ children }) => {
```
FC interface already includes `{children}`